src/components/table-freeze.vue

```diff
@@ -0,0 +1,204 @@
+https://www.apache.org/licenses/LICENSE-2.0. No part of ODK Central,
+including this file, may be copied, modified, propagated, or distributed
+except according to the terms contained in the LICENSE file.
+-->
+<template>
+  <div class="table-freeze">
+    <table :id="idsSplit[0]" class="table table-freeze-frozen"
+      :class="{ divider }">
+      <thead>
+        <tr>
+          <slot name="head-frozen"></slot>
+        </tr>
+      </thead>
+      <!-- eslint-disable-next-line vuejs-accessibility/click-events-have-key-events -->
+      <tbody v-if="data != null" ref="frozenBody"
+        :class="`actions-trigger-${actionsTrigger}`"
+        @mousemove="setActionsTrigger('hover')"
+        @focusin="setActionsTrigger('focus')" @click="actionClick">
+        <slot v-for="(element, index) in data" :key="element[keyProp]"
+          name="data-frozen" :data="element" :index="index">
+        </slot>
+      </tbody>
+    </table>
+    <div v-if="!frozenOnly" class="table-freeze-scrolling-container">
+      <table :id="idsSplit[1]" class="table table-freeze-scrolling">
+        <thead>
+          <tr>
+            <slot name="head-scrolling"></slot>
+          </tr>
+        </thead>
+        <!-- eslint-disable-next-line vuejs-accessibility/mouse-events-have-key-events -->
+        <tbody v-if="data != null" ref="scrollingBody"
+          @mousemove="setActionsTrigger('hover')" @mouseover="toggleHoverClass"
+          @mouseleave="removeHoverClass">
+          <slot v-for="(element, index) in data" :key="element[keyProp]"
+            name="data-scrolling" :data="element" :index="index">
+          </slot>
+        </tbody>
+      </table>
+    </div>
+  </div>
+</template>
+
+<script setup>
+import { computed, ref, shallowRef, watch } from 'vue';
+
+// We may render many rows, so this component makes use of event delegation and
+// other optimizations.
+
+const props = defineProps({
+  data: Array,
+  keyProp: {
+    type: String,
+    required: true
+  },
+  frozenOnly: Boolean,
+  ids: String,
+  divider: Boolean
+});
+const emit = defineEmits(['action']);
+
+const idsSplit = computed(() =>
+  (props.ids != null ? props.ids.split(' ') : []));
+
+/*
+Actions are shown for a row if the cursor is over the row or if one of the
+actions is focused. However, it is possible for the cursor to be over one row
+while an action is focused in a different row. In that case, we show the actions
+for one of the two rows depending on the type of the most recent event.
+
+I tried other approaches before landing on this one. However, some sequences of
+events were a challenge. For example, in SubmissionTable:
+
+  - Click the More button for a row.
+  - Next, press tab to focus the Review button in the next row.
+  - Actions are shown for the next row and are no longer shown beneath the
+    cursor. However, that will trigger a mouseover event, which depending on the
+    approach may cause actions to be shown beneath the cursor again.
```

Comment:
Ah interesting! This seems to happen when mouse and keyboard tabbing are used together. For example, if you tab to buttons on the 5th row, then move your mouse over the 2nd row, the buttons on the 5th row will go away, but the tooltip is still shown. I think this is an existing bug that we introduced with tooltips, not a regression in this PR. What's happening is that the buttons on the 5th row aren't being unmounted, but being moved offscreen. If they were unmounted, the tooltip would be removed automatically, but it's not as easy to hook into "moved offscreen," because that's driven by CSS. From the perspective of `v-tooltip`, the button on the 5th row still has focus, so there's no reason to remove the tooltip (tabbing would focus the More button on the 5th row).

I think it's possible to cause an orphaned tooltip even without showing another tooltip. In my example above, if you move your mouse over the 2nd row, but not over the edit button, you see that case. So I think there should be a more general change like "hide a tooltip if its anchor element goes out of view". We could maybe use an intersection observer for that. Or "hide a tooltip if the position of its anchor element changes". I think I've noticed a tooltip bug related to scrolling the page that that would also fix. How about I file a separate issue along these lines?

Reply:
Agree with your suggestion to handle this separately.

Reply:
Filed at #814.